geode-core/src/main/java/org/apache/geode/internal/security/shiro/JMXShiroAuthenticator.java

```diff
@@ -72,12 +73,17 @@ public Subject authenticate(Object credentials) {
   }
 
   @Override
-  public void handleNotification(Notification notification, Object handback) {
+  public synchronized void handleNotification(Notification notification, Object handback) {
```

Comment:
I agree with Jinmei that this will probably not work right based on the fact that there is no way to differentiate users. It might work better if you maintain a hash set of connectionIds.